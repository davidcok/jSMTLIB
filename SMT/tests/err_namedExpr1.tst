; testing that named expr are in scope
(set-logic QF_UF)
(declare-fun p () Bool)
(declare-fun q () Bool)
(assert (! p :named q)) ; conflicts with existing name
(assert (! p :named P)); OK
(assert (! q :named P)) ; conflicts
(declare-fun P () Bool) ; conflicts
(assert (and (! p :named PP) (! q :named P))) ; second conflicts, first should be withdrawn
(assert PP) ; should not exist
