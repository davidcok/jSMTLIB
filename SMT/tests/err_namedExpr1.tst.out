success
success
success
(assert (! p :named q)) ; conflicts with existing name
                    ^
(error "Symbol q is already defined")
success
(assert (! q :named P)) ; conflicts
                    ^
(error "Symbol P is already defined")
(declare-fun P () Bool) ; conflicts
             ^
(error "Symbol P is already defined")
(assert (and (! p :named PP) (! q :named P))) ; second conflicts, first should be withdrawn
                                         ^
(error "Symbol P is already defined")
(assert PP) ; should not exist
        ^^
(error "Unknown constant symbol PP")
